lib/ts/recipe/passwordless/recipeImplementation.ts

```diff
@@ -0,0 +1,227 @@
+          flowType: FlowType;
+      }
+    | {
+          status: "GENERAL_ERROR";
+          message: string;
+      };
+
+type ResendCodeApiResponse = {
+    status: "OK" | "RESTART_FLOW_ERROR";
+};
+
+type ConsumeCodeApiResponse =
+    | {
+          status: "OK";
+          createdUser: boolean;
+          user: {
+              id: string;
+              email?: string;
+              phoneNumber?: string;
+          };
+      }
+    | {
+          status: "INCORRECT_USER_INPUT_CODE_ERROR" | "EXPIRED_USER_INPUT_CODE_ERROR";
+          failedCodeInputAttemptCount: number;
+          maximumCodeInputAttempts: number;
+      }
+    | { status: "GENERAL_ERROR"; message: string }
+    | { status: "RESTART_FLOW_ERROR" };
+
+type ExistsAPIResponse = {
+    status: "OK";
+    exists: boolean;
+};
+
+export default function getRecipeImplementation(recipeId: string, appInfo: NormalisedAppInfo): RecipeInterface {
+    const querier = new Querier(recipeId, appInfo);
+
+    return {
+        createCode: async function (
+            input: ({ email: string } | { phoneNumber: string } | { deviceId: string; preAuthSessionId: string }) & {
+                config: NormalisedConfig;
+            }
+        ): Promise<CreateCodeApiResponse> {
+            let bodyObj;
+            if ("email" in input) {
+                const validationRes = await input.config.validateEmailAddress(input.email);
+                if (validationRes !== undefined) {
+                    return {
+                        status: "GENERAL_ERROR",
+                        message: validationRes,
+                    };
+                }
+                bodyObj = {
+                    email: input.email,
+                };
+            }
+            if ("phoneNumber" in input) {
+                const validationRes = await input.config.validatePhoneNumber(input.phoneNumber);
+                if (validationRes !== undefined) {
+                    return {
+                        status: "GENERAL_ERROR",
+                        message: validationRes,
+                    };
+                }
+                bodyObj = {
+                    phoneNumber: input.phoneNumber,
+                };
+            }
+
+            return await querier.post("/signinup/code", { body: JSON.stringify(bodyObj) }, (context) => {
+                return input.config.preAPIHook({
+                    ...context,
+                    action: "PASSWORDLESS_CREATE_CODE",
+                });
+            });
+        },
+        resendCode: async function (
+            input: { deviceId: string; preAuthSessionId: string } & {
+                config: NormalisedConfig;
+            }
+        ): Promise<ResendCodeApiResponse> {
+            const bodyObj = {
+                deviceId: input.deviceId,
+                preAuthSessionId: input.preAuthSessionId,
+            };
+
+            return await querier.post("/signinup/code/resend", { body: JSON.stringify(bodyObj) }, (context) => {
```

Comment:
You need to check if this was successful. If it was, you need to fire the relevant event using `onHandleEvent`